phys2bids/physio_obj.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+I/O objects for phys2bids.
+"""
+
+import numpy as np
+
+
+def is_valid(var, var_type, list_type=None, return_var=True):
+    """
+    Checks that the var is of a certain type.
+    If type is list and list_type is specified,
+    checks that the list contains list_type.
+    Input
+    -----
+    var:
+        Variable to be checked.
+    var_type: type
+        Type the variable is assumed to be.
+    list_type: type
+        As type.
+
+    Output
+    ------
+    var:
+        Variable to be checked (same as input).
+    """
+    if not isinstance(var, var_type):
+        raise AttributeError('Something went wrong while populating blueprint')
+
+    if var_type is list and list_type is not None:
+        for element in var:
+            is_valid(element, list_type, return_var=False)
+
+    if return_var:
+        return var
+
+
+def has_size(var, data_size, token):
+    """
+    Checks that the var has the same dimension of the data
+    If it's not the case, fill in the var or removes exceding var entry.
+    Input
+    -----
+    var:
+        Variable to be checked.
```

Comment:
What is the type to expect? Is it a numpy array?

Reply:
Also, you should indicate that the shape of the array (if it's an array) is `data_size x 1`.

Reply:
I see that `var` is always a list.

Reply:
var can actually be anything, at the moment it's either a list or a numpy array. I can add that. The shape changes based on the input

Reply:
Should I write "any" for types?

Reply:
I would write `list or numpy array` so then we now what the can be. Specially because you're appending stuff to `var` and this could be a source of conflicts.

Reply:
That was my point. We should make sure that it works with different types if that's what we're going for. I would leave it for a future PR anyway.